common/rpc/interceptor/telemetry.go

```diff
@@ -361,21 +364,68 @@ func (ti *TelemetryInterceptor) streamMetricsHandlerLogTags(
 	), []tag.Tag{tag.Operation(overridedMethodName)}
 }
 
-func (ti *TelemetryInterceptor) HandleError(
-	req interface{},
+func (ti *TelemetryInterceptor) HandleError(req interface{},
 	metricsHandler metrics.Handler,
 	logTags []tag.Tag,
 	err error,
-) {
-
+	nsName namespace.Name) {
 	metrics.ServiceErrorWithType.With(metricsHandler).Record(1, metrics.ServiceErrorTypeTag(err))
 
 	if common.IsContextDeadlineExceededErr(err) || common.IsContextCanceledErr(err) {
 		return
 	}
 
+	statusCode := serviceerror.ToStatus(err).Code()
+	logAllErrors := nsName != "" && ti.logAllReqErrors(nsName.String())
+	if !logAllErrors && isUserCaused(statusCode) {
+		return
```

Comment:
nit: I personally feel the early return here makes the logic harder to understand as it means we may skip the recordMetric call later. I'd probably separate log and metric logic into two func and always invoke both.